src/daemon/distribution.c

```diff
@@ -0,0 +1,148 @@
+#include <math.h>
+#include "collectd.h"
+#include "distribution.h"
+
+struct bucket_s{
+	size_t bucket_counter;
+	double min_boundary;
+	double max_boundary;
+};
+
+struct distribution_s {
+     bucket_t *buckets;
+     size_t num_buckets;
+     size_t total_scalar_count;
+     double raw_data_sum;
+};
+
+//bucket constructor
+bucket_t initialize_bucket(double min_boundary, double max_boundary) {
+    bucket_t new_bucket = {
+        .bucket_counter = 0,
+        .min_boundary = min_boundary,
+        .max_boundary = max_boundary,
+    };
+    return new_bucket;
+} 
+
+//increment bucket_counter of specific bucket
+void increment(bucket_t *bucket) { 
```

Comment:
This function appears to be unused. I think you can safely remove it.

Reply:
Oh right. I removed it now.